client/starwhale/core/runtime/process.py

```diff
@@ -72,19 +73,36 @@ def run(self) -> None:
         for p in clear_positions[::-1]:
             argv.pop(p)
 
-        sub_cmd = " ".join(argv)
+        return self.run_arbitrary_cmd(argv)
+
+    run = run_self_swcli_cmd
+
+    def run_arbitrary_cmd(
+        self,
+        cmd: str | t.List[str],
+        cwd: str | Path | None = None,
+        live_stream: bool = False,
+    ) -> None:
+        """Run the arbitrary command in the runtime environment."""
+        if os.environ.get(self.EnvInActivatedProcess, "0") == "1":
+            raise RuntimeError("already in runtime activated process")
+
+        if isinstance(cmd, (list, tuple)):
+            cmd = " ".join([f"'{c}'" for c in cmd])
 
         # TODO: support windows platform
         if self._mode == PythonRunEnv.VENV:
             _bin_path = (self._prefix_path / "bin/activate").absolute()
-            cmd = " && ".join([f"source {_bin_path}", sub_cmd])
+            cmd = " && ".join([f"source {_bin_path}", cmd])
         elif self._mode == PythonRunEnv.CONDA:
-            cmd = f"{get_conda_bin()} run --live-stream --prefix {self._prefix_path.absolute()} {sub_cmd}"
+            cmd = f"{get_conda_bin()} run --live-stream --prefix {self._prefix_path.absolute()} {cmd}"
```

Comment:
are there should judge the param?
```
if live_stream:
  cmd = f"{get_conda_bin()} run --live-stream --prefix {self._prefix_path.absolute()} {cmd}" 
elif
  cmd = f"{get_conda_bin()} run --prefix {self._prefix_path.absolute()} {cmd}"
```

Reply:
no need. `--live-stream` is only used for starwhale popen function.